shared/util.go

```diff
@@ -214,3 +215,13 @@ func StringSliceContains(ss []string, s string) bool {
 	}
 	return false
 }
+
+// MapStringKeys returns the keys in the given string-keyed map.
+func MapStringKeys(m interface{}) []string {
+	keys := reflect.ValueOf(m).MapKeys()
```

Comment:
nit: Consider checking `reflect.ValueOf(m).Kind()` and returning `([]string, error)` to avoid panic if `m` is not a map.